src/dso_api/dynamic_api/remote/serializers.py

```diff
@@ -63,11 +96,25 @@ def get_schema(self, instance):
         dataset_path = Dataset.objects.get(name=name).path
         return f"https://schemas.data.amsterdam.nl/datasets/{dataset_path}/dataset#{table}"
 
+    def _lookup_field(self, id):
+        return self.table_schema.get_field_by_id(id)
+
+
+class RemoteFieldSerializer(DSOSerializer, _AuthMixin):
+    """Serializer for nested objects in the schema.
+    This can be a field in RemoteSerializer instances."""
 
-class RemoteObjectSerializer(DSOSerializer):
     table_schema = None  # defined by factory
     field_schema = None  # defined by factory
 
+    def get_fields(self):
+        return self._filter_authorized_fields(super().get_fields())
+
+    def _lookup_field(self, id):
+        for field in self.field_schema.sub_fields:
+            if field.id == id:
+                return field
```

Comment:
Voor nu prima, maar zou je hier niet een aparte functie voor maken in schematools? `self.field_schema.get_sub_field_by_id(..)` ?

Reply:
Tijdelijke oplossing. Beide projecten tegelijk aanpassen geeft altijd zo'n gedoe.